packages/restate-sdk/src/context_impl.ts

```diff
@@ -188,17 +210,32 @@ export class ContextImpl implements ObjectContext, WorkflowContext {
           : []
       );
 
-      // TODO eventually we return this promise back to the user
-      // const invocationIdPromise = this.createInvocationIdPromise(
-      //   call_handles.invocation_id_completion_id
-      // );
+      const invocationIdHandle = call_handles.invocation_id_completion_id;
+      const invocationHandle = () => {
+        // create a promise, on demand that resolves to an invocation id if needed.
+        return this.createInvocationIdPromise(invocationIdHandle);
+      };
```

Comment:
This is wrong, because it has the side effect that once you await the invocation id of a call once, you cannot await it again. I think it's better to create the promise outside the closure, so that the user in case it calls twice invocationId(), it gets the same promise.

```suggestion
      const p = this.createInvocationIdPromise(invocationIdHandle)
      const invocationHandle = () => {
        // create a promise, on demand that resolves to an invocation id if needed.
        return p;
      };
```

Reply:
> how about caching it?

This works, as long as you give to the user the promise you created the first time.